app/services/concerns/xccdf/util.rb

```diff
@@ -52,10 +52,25 @@ def set_openscap_parser_data
 
       def invalidate_cache
         Rails.cache.delete("#{@host&.id}/failed_rule_objects_result")
-        Rails.cache.delete("#{@new_host_profile&.id}/#{@host&.id}/results")
+        Rails.cache.delete("#{@host_profile&.id}/#{@host&.id}/results")
+        Rails.cache.delete("#{@host_profile&.id}/rules")
+        Rails.cache.delete("#{@host.id}/#{@host_profile&.id}/rules_passed")
+        Rails.cache.delete("#{@host.id}/#{@host_profile&.id}/rules_failed")
+        Rails.cache.delete("#{@host.id}/#{@host_profile&.id}/score")
         @host_profile.rules.each do |rule|
```

Comment:
Wow, maybe we could break this into a method, like `invalidate_cache_for(host:, profile:)`